src/Parsers/FunctionSecretArgumentsFinder.h

```diff
@@ -623,7 +623,7 @@ class FunctionSecretArgumentsFinder
             /// PostgreSQL('host:port', 'database', 'user', 'password')
             findMySQLDatabaseSecretArguments();
         }
-        else if (engine_name == "S3")
+        else if (engine_name == "S3" || engine_name == "DataLakeCatalog")
         {
             /// S3('url', 'access_key_id', 'secret_access_key')
             findS3DatabaseSecretArguments();
```

Comment:
So it is not a `S3` database secret anymore :)

Reply:
yes. but `DataLakeCatalog` should (and will) support azure as well. So it's only a hot-fix

Reply:
Probably it's better to create another function like `findS3DatabaseSecretArguments()` or rename it.

Reply:
created a new function and left comments, PTAL